contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/VoldemortBuildAndPushJob.java

```diff
@@ -830,7 +862,13 @@ public void runPushStore(Props props, String url, String dataDir) throws Excepti
                 maxNodeFailures,
                 failedFetchStrategyList,
                 url,
+                modifiedDataDir,
                 buildPrimaryReplicasOnly).run();
+
+        if (distcpJob != null && distcpJob.getCdnTargetFS() != null) {
+            // This would allow temp dirs marked by deleteDirOnExit() to be deleted.
+            distcpJob.getCdnTargetFS().close();
```

Comment:
I prefer to call 'distcpJob.closeCdnTargetFS()', which will internally check whether the filesystem is null or not, and close it if not.
Generally, external objects should not operate the fields owned by some other object to avoid tight coupling.